flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/utils/WindowEmitStrategy.scala

```diff
@@ -153,12 +152,22 @@ object WindowEmitStrategy {
     }
     val enableEarlyFireDelay = tableConfig.getConfiguration.getBoolean(
       TABLE_EXEC_EMIT_EARLY_FIRE_ENABLED)
-    val earlyFireDelay = getMillisecondFromConfigDuration(
-      tableConfig, TABLE_EXEC_EMIT_EARLY_FIRE_DELAY)
+    val earlyFireDelay: JLong = if (tableConfig.getConfiguration
+      .getOptional(TABLE_EXEC_EMIT_EARLY_FIRE_DELAY).isPresent) {
+      tableConfig.getConfiguration
+        .get(TABLE_EXEC_EMIT_EARLY_FIRE_DELAY).toMillis
+    } else {
+      null
+    }
```

Comment:
What about refactoring the `earlyFireDelay` to `Duration` type? Then we can simplify the above code to

```suggestion
val earlyFireDelay: Duration = tableConfig.getConfiguration
      .getOptional(TABLE_EXEC_EMIT_EARLY_FIRE_DELAY)
      .orElse(null)
```

I think `Duration` is more concise than `Long`.

Reply:
Yes, Duration type for earlyFireDelay and lateFireDelay will be more reasonable.
I have updated, thanks for reviewing again.